### DISCUSSION TITLE
May I know the file that manages the code blocks?

### DISCUSSION THREAD

**Discussion by drj9812** [cotes2020/jekyll-theme-chirpy#1752] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Built from `jekyll-theme-chirpy`

### Description

What files manage code blocks? I don't think bash style code blocks are supported, but I'd like to add them.

### Operations you have already tried

Search on google.

### Anything else?

no.

**Comment by huanyushi** [comment, marked as answer]
See [here](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_config.yml#L157), chirpy use [rouge](https://github.com/rouge-ruby/rouge) to highlight code. And the list of supported languages and lexers can be seen [here](https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers).

If you want to change the color, relevent settings can be seen [syntax-dark](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_sass/colors/syntax-dark.scss) and [syntax-light](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_sass/colors/syntax-light.scss).

Run `rougify help style` to get the avaliable themes list:

`base16, base16.dark, base16.light, base16.monokai, base16.monokai.dark, base16.monokai.light, base16.solarized, base16.solarized.dark, base16.solarized.light, bw, colorful, github, github.dark, github.light, gruvbox, gruvbox.dark, gruvbox.light, igorpro, magritte, molokai, monokai, monokai.sublime, pastie, thankful_eyes, tulip`

You can choose one of themes to get the css file, and cover the original scss file. For example, you can change the code highlighting theme in dark mode to github.dark

<table>
<tr>
   <th>base16.dark</th>
   <th>github.dark</th>
</tr>
<tr>
<td><img src="https://github.com/cotes2020/jekyll-theme-chirpy/assets/100574401/9f53cedc-7739-4866-aa64-8fb0800ca37b" alt="base16.dark"></td>
<td><img src="https://github.com/cotes2020/jekyll-theme-chirpy/assets/100574401/a5532eda-3f4b-45a0-b239-6b345926149f" alt="github.dark"></td>
</tr>
</table>

Of course, you can also manually adjust the color yourself and design your own favorite theme.

**Reply by drj9812** [reply]
Thank you so much.

However, what I was talking about was whether it was possible to add code blocks in the bash(Git bash) language, not CSS.

Like, \```bash code...```.

Currently, the Chirpy theme does not seem to support \```bash code...```.

**Reply by huanyushi** [reply]
In my previous answer, I mentioned what languages rougy supports highlighting. You can find

> shell: Various shell languages, including sh and bash [aliases: bash,zsh,ksh,sh]

So I think you should use \```shell\``` instead of \```bash\```. This is exactly what I want to show you in the figures in my answer.

**Reply by drj9812** [reply]
So there's no way to get "bash" displayed.

Thank you. You've been a great help.

**Reply by huanyushi** [reply]
@drj9812 

> shell: Various shell languages, including sh and **bash** [aliases: bash,zsh,ksh,sh]

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/100574401/6201decd-fa1c-46c0-98f9-0bae26a8bc35)
